plugins/fileservices/fileservices.cpp

```diff
@@ -3128,3 +3128,49 @@ FILESERVICES_API void FILESERVICES_CALL fsClearExpireDays(ICodeContext * ctx, co
     else
         throw makeStringExceptionV(0, "ClearExpireDays: Could not find logical file %s", lfn.str());
 }
+
+bool getDefaultValue(const char * processName, const char * propertyName, StringBuffer & strDefaultValue)
+{
+    Owned<IConstEnvironment> daliEnv = openDaliEnvironment();
+    Owned<IPropertyTree> env = getEnvironmentTree(daliEnv);
+
+    if (env.get())
+    {
+        Owned<IPropertyTreeIterator> processIter = env->getElements(processName);
+        if (processIter->first())
+        {
+            if (processIter->query().hasProp(propertyName))
+            {
+                processIter->query().getProp(propertyName, strDefaultValue);
+                return true;
+            }
+        }
+    }
+    return false;
+}
+
+FILESERVICES_API bool FILESERVICES_CALL fsGetNoCommonDefault()
+{
+    // This function returns the value of the DFUServerProcess' 'noCommon' property, if it is exists. 
+    // Othervise it returns with true, the recent default value of 'noCommon'
+
+    StringBuffer strDefaultValue;
+    if (getDefaultValue("Software/DfuServerProcess", "@noCommon", strDefaultValue))
+        return strToBool(strDefaultValue.str());
+    return true;
+}
+
+FILESERVICES_API int FILESERVICES_CALL fsGetMaxConnectionsDefault()
```

Comment:
Why is this needed, rather than dfuserver using a global default value?  It is already only set in the dfu request if it has been specified.

Reply:
@jakesmith ?

Reply:
Agree, for this and any other property that can, it should pickup the default within dfuserver when it's processing the dfu workunit.
NB: it already defaults there to a hard-wired DEFAULT_MAX_CONNECTIONS, so that can be switched for the global default if configured.

Unfortunately, the same cannot be applied to boolean flags (e.g. noCommon) where it's default can not be a 'unset' value and therefore cannot be tested in dfuserver in order to use a global default.
For properties that can't (e.g. noCommon), the current approach of picking up a configured global default and passing it in, is the best approach I can think of.